tools/scenario-player/scenario_player/scenario.py

```diff
@@ -0,0 +1,313 @@
+    @property
+    def raiden_version(self):
+        return self._config.get('raiden_version', 'LATEST')
+
+    @property
+    def count(self):
+        try:
+            return self._config['count']
+        except KeyError:
+            raise MissingNodesConfiguration('Must specify a "count" setting!')
+
+    @property
+    def default_options(self):
+        return self._config.get('default_options', {})
+
+    @property
+    def node_options(self):
+        return self._config.get('node_options', {})
+
+    @property
+    def nodes(self) -> List[str]:
+        """Return the list of nodes configured in the scenario's yaml.
+
+        Should the scenario use version 1, we check if there is a 'setting'.
+        If so, we derive the list of nodes from this dictionary, using its
+        'first', 'last' and 'template' keys. Should any of these keys be
+        missing, we throw an appropriate exception.
+
+        If the scenario version is not 1, or no 'range' setting exists, we use
+        the 'list' settings key and return the value. Again, should the key be
+        absent, we throw an appropriate error.
+
+        :raises MissingNodesConfiguration:
+            if the scenario version is 1 and a 'range' key was detected, but any
+            one of the keys 'first', 'last', 'template' are missing; *or* the
+            scenario version is not 1 or the 'range' key and the 'list' are absent.
+        :rtype: List
+        """
+        if self._scenario_version == 1 and 'range' in self._config:
+            range_config = self._config['range']
+
+            try:
+                start, stop = range_config['first'], range_config['last'] + 1
+            except KeyError:
+                raise MissingNodesConfiguration(
+                    'Setting "range" must be a dict containing keys "first" and "last",'
+                    ' whose values are integers!'
+                )
+
+            try:
+                template = range_config['template']
+            except KeyError:
+                raise MissingNodesConfiguration(
+                    'Must specify "template" setting when giving "range" setting.'
+                )
+
+            return [template.format(i) for i in range(start, stop)]
+        try:
+            return self._config['list']
+        except KeyError:
+            raise MissingNodesConfiguration('Must specify nodes under "list" setting!')
+
+    @property
+    def commands(self) -> Dict:
+        """Return the commands configured for the nodes.
+
+        :rtype: Dict
+        """
+        return self._config.get('commands', {})
+
+
+class Scenario(Mapping):
+    """Thin wrapper class around a scenario .yaml file.
+
+    Handles default values as well as exception handling on missing settings.
+
+    :param pathlib.Path yaml_path: Path to the scenario's yaml file.
+    """
+    def __init__(self, yaml_path: pathlib.Path) -> None:
+        self._yaml_path = yaml_path
+        self._config = yaml.load(yaml_path.open())
```

Comment:
This should be `yaml.safe_load()` (And yes, I know you only copied that from what I originally wrote ;))

Reply:
tsk, tsk, tsk. Done. ;)